kyo-direct/shared/src/main/scala/kyo/internal/AsyncShift.scala

```diff
@@ -17,9 +18,162 @@ end asyncShiftLowPriorityImplicit1
 object asyncShift extends asyncShiftLowPriorityImplicit1:
 
     transparent inline given shiftedChunk[A]: ChunkAsyncShift[A] = new ChunkAsyncShift[A]
+    transparent inline given shiftedMaybe: MaybeAsyncShift       = new MaybeAsyncShift
+    transparent inline given shiftedResult: ResultAsyncShift     = new ResultAsyncShift
 
 end asyncShift
 
+class ResultAsyncShift(using Frame) extends AsyncShift[Result.type]:
+
+    def map[F[_], E, A](result: Result.type, monad: CpsMonad[F])(ma: Result[E, A])[B](f: A => F[B]): F[Result[E, B]] =
+        monad match
+            case _: KyoCpsMonad[?] => ma match
+                    case Result.Success(a) => f(a).map(Result.Success(_))
```

Comment:
It seems to still be missing the handling of Panic?

Reply:
Indeed, sorry, will write some tests for panic.